### ISSUE TITLE
Addditional OkHttp3Downloader constructor with OkHttpClient.Builder()

### ISSUE THREAD

**Issue by flasher297** [JakeWharton/picasso2-okhttp3-downloader#3]
There is no ability to set custom OkHttpClient but with builtin OkHttp3Downloader cache.
It would be handy to have such constructor and private method.
```
public OkHttp3Downloader(OkHttpClient.Builder okhttpBuilder, Context) {
    File cacheDir = createDefaultCacheDir(context);
    this(defaultCacheOkHttpClient(okhttpBuilder, cacheDir, calculateDiskCacheSize(cacheDir)));
  }

 private static OkHttpClient defaultCacheOkHttpClient(OkHttpClient.Builder okhttpBuilder, File cacheDir, long maxSize {
         okhttpBuilder
        .cache(new okhttp3.Cache(cacheDir, maxSize))
        .build();
  }
```
Usage
```
OkHttpClient.Builder defaultHttpClientBuilder = new OkHttpClient.Builder()
            .retryOnConnectionFailure(false)
            .readTimeout(40, TimeUnit.SECONDS)
            .connectTimeout(40, TimeUnit.SECONDS);
OkHttp3Downloader okHttp3Downloader = new OkHttp3Downloader(defaultHttpClientBuilder, this)
```
Or maybe createDefaultCacheDir() methoud could be public?

**Comment by JakeWharton** [comment]
Let's make the method public.

**Comment by JakeWharton** [comment]
Actually maybe I'll just make a new method that produces a `Cache`?

**Comment by flasher297** [comment]
@JakeWharton Thank you for including this fix in 1.1.0 release.
This's what I need. Now added custom OkHttpClient with build in OkHttp3Downloader cache =)
